### PR TITLE
Migrated to native jakarta

### PR DESCRIPTION
**Pull request by rsearls**
Import changes to reference jakarta name space.
Referencing org.apache.santuario:xmlsec:2.3.1-SNAPSHOT because no final version is available.

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
         <wsdl4j.version>1.6.3</wsdl4j.version>
         <xalan.version>2.7.2</xalan.version>
         <xz.version>1.9</xz.version>
-        <xmlsec.version>2.3.0</xmlsec.version>
+        <xmlsec.version>2.3.1-SNAPSHOT</xmlsec.version>
```
Comment (coheigea) [inline parent/pom.xml:50]: Change to 3.0.0-SNAPSHOT after applying https://github.com/apache/santuario-xml-security-java/pull/63

I see some errors:

java.lang.ExceptionInInitializerError
	at org.apache.wss4j.stax.test.VulnerabliltyVectorsDecompressedBytesTest.setup(VulnerabliltyVectorsDecompressedBytesTest.java:44)
Caused by: java.lang.RuntimeException: Implementation of Jakarta XML Binding-API has not been found on module path or classpath.

**Review by coheigea — CHANGES_REQUESTED** [review]

File: parent/pom.xml (hunk 1 above)
Comment (rsearls) [reply parent/pom.xml]: Requested change made.
Test failures resolved.

File: ws-security-stax/pom.xml [hunk 2]
```diff
@@ -103,6 +107,10 @@
             <artifactId>bcprov-jdk15on</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.sun.xml.bind</groupId>
+            <artifactId>jaxb-impl</artifactId>
```
Comment (coheigea) [inline ws-security-stax/pom.xml:112]: Could we change all instances of this dependency to:

<groupId>org.glassfish.jaxb</groupId>
<artifactId>jaxb-runtime</artifactId>

**Review by coheigea — CHANGES_REQUESTED** [review]

File: parent/pom.xml [hunk 3]
```diff
@@ -56,10 +56,24 @@
         <wss4j.osgi.export>org.apache.wss4j.*</wss4j.osgi.export>
         <wss4j.osgi.dynamic.import />
         <wss4j.manifest.location>${project.build.outputDirectory}/META-INF/MANIFEST.MF</wss4j.manifest.location>
+        <jakarta.xml.bind-api.version>3.0.1</jakarta.xml.bind-api.version>
+        <jakarta.mail.version>2.0.0</jakarta.mail.version>
+        <jaxb-runtime.version>3.0.2</jaxb-runtime.version>
```
Comment (coheigea) [inline parent/pom.xml:61]: There are a bunch of old JAXB dependencies in the root pom under the JDK9 profile that need to be removed:

<dependency>
                     <groupId>jakarta.xml.ws</groupId>
                     <artifactId>jakarta.xml.ws-api</artifactId>
                     <version>2.3.3</version>
                 </dependency>
                 <dependency>
                     <groupId>jakarta.xml.bind</groupId>
                     <artifactId>jakarta.xml.bind-api</artifactId>
                     <version>2.3.3</version>
                 </dependency>
                 <dependency>
                     <groupId>org.glassfish.jaxb</groupId>
                     <artifactId>jaxb-runtime</artifactId>
                     <version>2.3.2</version>
                     <scope>test</scope>
                 </dependency>
Once these are removed, then you can remove the jakarta API from ws-security-stax etc., as it is added already as a compile time dependency in Santuario (I changed it this morning so you might need to reinstall it).

**Review by coheigea — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -394,22 +394,6 @@
...
-                     <version>2.3.3</version>
-                 </dependency>
-                 <dependency>
-                     <groupId>org.glassfish.jaxb</groupId>
-                     <artifactId>jaxb-runtime</artifactId>
-                     <version>2.3.2</version>
-                     <scope>test</scope>
-                 </dependency>
                  <dependency>
                      <groupId>com.sun.xml.messaging.saaj</groupId>
```
Comment (coheigea) [inline pom.xml:398]: This needs to be updated to 2.0.1 and some compile time dependencies fixed:

[ERROR] /Users/coheigea/src/apache/ws-wss4j/ws-security-common/src/test/java/org/apache/wss4j/common/util/SOAPUtil.java:[24,22] package javax.xml.soap does not exist
[ERROR] /Users/coheigea/src/apache/ws-wss4j/ws-security-common/src/test/java/org/apache/wss4j/common/util/SOAPUtil.java:[45,20] cannot find symbol
  symbol:   class MessageFactory
  location: class org.apache.wss4j.common.util.SOAPUtil

**Review by coheigea — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 4 above)
Comment (rsearls) [reply pom.xml:398]: Done

File: parent/pom.xml [hunk 5]
```diff
@@ -56,10 +56,29 @@
         <wss4j.osgi.export>org.apache.wss4j.*</wss4j.osgi.export>
         <wss4j.osgi.dynamic.import />
         <wss4j.manifest.location>${project.build.outputDirectory}/META-INF/MANIFEST.MF</wss4j.manifest.location>
+        <jakarta.xml.bind-api.version>3.0.1</jakarta.xml.bind-api.version>
+        <jakarta.xml.soap-api.version>2.0.1</jakarta.xml.soap-api.version>
```
Comment (coheigea) [inline parent/pom.xml:60]: Remove

File: parent/pom.xml [hunk 6]
```diff
@@ -56,10 +56,29 @@
...
+            <dependency>
+                <groupId>jakarta.xml.bind</groupId>
+                <artifactId>jakarta.xml.bind-api</artifactId>
+                <version>${jakarta.xml.bind-api.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>jakarta.xml.soap</groupId>
+                <artifactId>jakarta.xml.soap-api</artifactId>
+                <version>${jakarta.xml.soap-api.version}</version>
+            </dependency>
```
Comment (coheigea) [inline parent/pom.xml:81]: Remove

**Review by coheigea — CHANGES_REQUESTED** [review]

File: parent/pom.xml (hunk 6 above)
Comment (rsearls) [reply parent/pom.xml]: org.glassfish.jaxb:jaxb-runtime can not be removed from ws-security-stax/pom.xml
it is required for KerberosTest.java to run successfully.

All other changes are complete.

Comment (coheigea) [reply parent/pom.xml]: Could it be changed to runtime scope instead? and finally, can you create a WSS JIRA for this change and reference the JIRA number in the commit message? Then we should be good to merge.

Comment (rsearls) [reply parent/pom.xml]: Change complete.  
Created https://issues.apache.org/jira/browse/WSS-694 Move wss4j to native jakarta namespace.  Added it to the PR commit msg.

**Review by coheigea — APPROVED** [review]
Thanks a lot for your contribution. One final check before merge - do you have a ICLA with Apache? https://www.apache.org/licenses/icla.pdf

**Comment by rsearls** [comment]
emailed

**Comment by coheigea** [comment]
Did you make a note to inform the webservices project?

**Comment by rsearls** [comment]
Yes, here is the jira
https://issues.apache.org/jira/browse/WSS-694 Move wss4j to native jakarta namespace

**Comment by rsearls** [comment]
Which webservices project are you thinking of?

**Comment by coheigea** [comment]
There is a "Notify Project" field in https://www.apache.org/licenses/icla.pdf, I was wondering if you added in WebServices (WSS4J parent project), to inform us. Otherwise I will try to find out directly if it was received.

**Review by coheigea — APPROVED** [review]

**Pull request merged**